packages/design-system/src/components/N8nUserStack/UserStack.vue

```diff
@@ -0,0 +1,181 @@
+<script lang="ts" setup>
+import type { IUser } from '@/types';
+import N8nAvatar from '../N8nAvatar';
+import N8nUserInfo from '../N8nUserInfo';
+import type { PropType } from 'vue';
+import { computed } from 'vue';
+
+const props = defineProps({
+	users: {
+		type: Object as PropType<{ [groupName: string]: IUser[] }>,
+		required: true,
+	},
+	currentUserEmail: {
+		type: String,
+		required: true,
+	},
+	maxAvatars: {
+		type: Number,
+		default: 2,
+		validator: (value: number) => value > 0,
+	},
+	dropdownTrigger: {
+		type: String,
+		default: 'hover',
+		validator: (value: string) => ['hover', 'click'].includes(value),
+	},
+});
+
+// Filter out empty groups
+const groups = computed(() => {
+	const users: { [groupName: string]: IUser[] } = {};
+
+	for (const groupName in props.users) {
+		if (props.users[groupName].length > 0) {
+			users[groupName] = props.users[groupName];
+		}
+	}
+
+	return users;
+});
+
+const groupCount = computed(() => {
+	return Object.keys(props.users).length;
+});
+
+const flatUserList = computed(() => {
+	const users: IUser[] = [];
+
+	for (const groupName in props.users) {
+		users.push(...props.users[groupName]);
+	}
+
+	return users;
+});
+
+const visibleAvatarCount = computed(() => {
+	return flatUserList.value.length >= props.maxAvatars
+		? props.maxAvatars
+		: flatUserList.value.length;
+});
+
+const hiddenUsersCount = computed(() => {
+	return flatUserList.value.length - visibleAvatarCount.value;
+});
+
+const menuHeight = computed(() => {
+	return groupCount.value > 1 ? 220 : 190;
+});
+</script>
+
+<template>
+	<div class="user-stack" data-test-id="user-stack-container">
+		<el-dropdown
+			:trigger="$props.dropdownTrigger"
+			:max-height="menuHeight"
+			popper-class="user-stack-popper"
+		>
+			<div :class="$style.avatars">
+				<n8n-avatar
+					v-for="user in flatUserList.slice(0, visibleAvatarCount)"
+					:key="user.id"
+					:firstName="user.firstName"
+					:lastName="user.lastName"
+					:class="$style.avatar"
+					size="small"
+				/>
+				<div v-if="hiddenUsersCount > 0" :class="$style.hiddenBadge">+{{ hiddenUsersCount }}</div>
+			</div>
+			<template #dropdown>
+				<el-dropdown-menu class="user-stack-list">
+					<div v-for="(groupUsers, index) in groups" :key="index">
+						<div v-if="groupUsers.length > 0" :class="$style.groupContainer">
```

Comment:
Isn't this always true since the `groups` doesn't contain empty groups?

Reply:
True, this is a leftover that I forgot to remove